vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java

```diff
@@ -1073,6 +1073,10 @@ public static VMwareDVSPvlanConfigSpec createDVPortPvlanConfigSpec(int vlanId, i
     }
 
     public static VmwareDistributedVirtualSwitchVlanSpec createDVPortVlanSpec(Integer vlanId, String vlanRange) {
+        if (vlanId != null && vlanId == 4095){
+            vlanId = null;
+            vlanRange = "0-4094";
+        }
```

Comment:
so `if (vlanId == 4095)` we set the range and  vlanId but `if (vlanId == null)` we don't care about the set range. is this intentional?

Reply:
@DaanHoogland Yes the 4095 vlan ID is used a magic no. to create a trunked port group; this already exists for other hypervisors like kvm - https://cwiki.apache.org/confluence/display/CLOUDSTACK/Enable+vlan+4095+%27tag+passthrough%27+for+KVM

Reply:
yes, i understand the function of `vlandId == 4095` but my point is that it will be possible to pass `vlanId == null` and `range == '2-9,15-99,4000-4094'` or even `range == '4000-4095'`, and should we correct that combination of parameters?

Reply:
maybe not relevant enough to this fix, never mind